module/zcommon/zfs_prop.c

```diff
@@ -772,6 +772,10 @@ zfs_prop_init(void)
 	    ZFS_TYPE_VOLUME, "<date>", "SNAPSHOTS_CHANGED", B_FALSE, B_TRUE,
 	    B_TRUE, NULL, sfeatures);
 
+	zprop_register_index(ZFS_PROP_LONGNAME, "longname", 0, PROP_INHERIT,
+	    ZFS_TYPE_FILESYSTEM, "on | off", "LONGNAME", boolean_table,
+	    sfeatures);
```

Comment:
I am somewhat split about this property and its disabled default.  I may speculate that some users may not want this functionality, while same time once it become routine I don't think I want to manually enable it on each new pool created.

Reply:
The reason it's not enabled is because it may not be suitable for all workload. For example NFS, or when you copy files between native linux fs.

Reply:
> I am somewhat split about this property and its disabled default. 
I am hoping to have a couple of days after it lands to see if I can make it work on my platforms, before it is made default ...